fix: Use []interface{} for protobuf marshaling in hybrid mode

tsanders-rh opened the pull request:
## Problem

Hybrid mode provider initialization fails for NodeJS, Python, and Go providers with the error:
```
proto: invalid type: []string
```

This prevents hybrid mode from working with any non-Java providers, making it impossible to analyze NodeJS, Python, or Go applications.

## Root Cause

In `cmd/analyze-hybrid.go`, the `setupNetworkProvider()` function builds provider configurations with typed slice literals (`[]string`):

```go
case util.GoProvider:
    providerSpecificConfig["workspaceFolders"] = []string{...}  // Line 237

case util.PythonProvider:
    providerSpecificConfig["workspaceFolders"] = []string{...}  // Line 243

case util.NodeJSProvider:
    providerSpecificConfig["lspServerArgs"] = []string{"--stdio"}  // Line 248
    providerSpecificConfig["workspaceFolders"] = []string{...}  // Line 249
```

These configurations are marshaled to protobuf when sending over gRPC to provider containers:

**Code path:**
1. `cmd/analyze-hybrid.go` calls `lib.GetProviderClient(providerConfig, log)`
2. For non-builtin providers, this returns `grpc.NewGRPCClient(config, log)`
3. `grpcProvider.Init()` calls `structpb.NewStruct(config.ProviderSpecificConfig)`
4. `structpb.NewStruct()` calls `NewValue()` for each map value
5. `NewValue()` only accepts `[]any` (i.e., `[]interface{}`), not typed slices

**Protobuf source code** (`google.golang.org/protobuf/types/known/structpb`):
```go
func NewValue(v any) (*Value, error) {
    switch v := v.(type) {
    case []any:  // ← ONLY accepts []interface{}, not []string
        v2, err := NewList(v)
        if err != nil {
            return nil, err
        }
        return NewListValue(v2), nil
    default:
        return nil, protoimpl.X.NewError("invalid type: %T", v)  // ← []string triggers this
    }
}
```

Go's type switch uses **exact type matching** - `[]string` will never match `case []any:`, so it falls through to the `default` case and returns an error.

## Why This Only Affects Hybrid Mode

| Mode | Providers | Config Marshaling | Affected? |
|------|-----------|-------------------|-----------|
| **Containerized** | In containers | Inside analyzer container | ❌ No |
| **Hybrid** | In containers | **On host** with protobuf v1.36.5 | ✅ **Yes** |
| **Containerless** | In-process | **Never happens** (direct calls) | ❌ No |

Hybrid mode is the first time kantra (running on the host) marshals provider configs to protobuf before sending to containerized providers. Containerless mode passes configs directly to in-process providers without marshaling.

## Solution

Changed all typed slice literals to `[]interface{}` in provider configurations:

```go
case util.GoProvider:
    providerSpecificConfig["workspaceFolders"] = []interface{}{...}

case util.PythonProvider:
    providerSpecificConfig["workspaceFolders"] = []interface{}{...}

case util.NodeJSProvider:
    providerSpecificConfig["lspServerArgs"] = []interface{}{"--stdio"}
    providerSpecificConfig["workspaceFolders"] = []interface{}{...}
```

This satisfies protobuf's type requirement while maintaining identical runtime behavior.

## Testing

Successfully tested with tackle2-ui NodeJS analysis:
- ✅ NodeJS provider initialized without errors
- ✅ Processed 582 TypeScript files
- ✅ Evaluated 57 PatternFly migration rules
- ✅ Generated output.yaml with violations found

**Before fix:**
```bash
$ kantra analyze --input ~/tackle2-ui --output ./out --run-local=false
unable to start provider nodejs: proto: invalid type: []string
```

**After fix:**
```bash
$ kantra analyze --input ~/tackle2-ui --output ./out --run-local=false
Running source analysis...
  ✓ Initialized providers (builtin, nodejs)
  ✓ Loaded 57 rules
Processing rules 100% |█████████████████████████| 57/57
Analysis complete!
```

## Impact

This fix enables hybrid mode to work with all provider types, not just Java. Without this fix, hybrid mode cannot analyze NodeJS, Python, Go, or .NET applications.

## References

- **Protobuf source**: https://github.com/protocolbuffers/protobuf-go/blob/master/types/known/structpb/struct.pb.go
- **gRPC provider**: https://github.com/konveyor/analyzer-lsp/blob/main/provider/grpc/provider.go

Fixes #624

shawn-hurley commented:
I think there is another place that provider configs exist, we should update there as well in pkg/provider I believe

tsanders-rh commented:
@shawn-hurley - Fixed! I've updated the PR to include the pkg/provider/ changes as well.

  Changes in this commit (b5a9c6d)

  Changed all typed slice literals to []interface{} in provider configurations:

  - pkg/provider/go.go:22 - workspaceFolders
  - pkg/provider/python.go:22 - workspaceFolders
  - pkg/provider/nodejs.go:22 - workspaceFolders
  - pkg/provider/nodejs.go:24 - lspServerArgs

  Why this matters:

  These configurations in pkg/provider/ are used for containerized mode where providers run in containers. When kantra starts these
  containerized providers, it calls GetConfigVolume() which builds provider configs that are marshaled to protobuf via gRPC.

  The same protobuf requirement applies here - structpb.NewStruct() requires []interface{} for slice values, not typed slices like
  []string.
